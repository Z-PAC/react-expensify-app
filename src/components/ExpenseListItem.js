import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

// load euro locale
numeral.register('locale', 'pt', {
  delimiters: {
    thousands: ' ',
    decimal: ','
  },
  abbreviations: {
    thousand: 'k',
    million: 'm',
    billion: 'b',
    trillion: 't'
  },
  currency: {
    symbol: 'â‚¬'
  }
});

numeral.locale('pt');

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>
      {numeral(amount/100).format('$0,0.00')} 
      - 
      {moment(createdAt).format('MMMM Do, YYYY')}
    </p>
  </div>
);

export default ExpenseListItem;