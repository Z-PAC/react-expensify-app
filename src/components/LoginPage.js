import React from 'react';
import { connect } from 'react-redux';
import { startLoginGoogle, startLoginFacebook } from '../actions/auth';

export const LoginPage = ({ startLoginGoogle, startLoginFacebook }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>Getting your expenses under control.</p>
      <div className="box-layout__data">
        <button className="button" onClick={startLoginGoogle}>Login with Google</button>
        <button className="button button--darkened" onClick={startLoginFacebook}>Login with Facebook</button>
      </div>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLoginGoogle: () => dispatch(startLoginGoogle()),
  startLoginFacebook: () => dispatch(startLoginFacebook())
});

export default connect (undefined,mapDispatchToProps)(LoginPage);