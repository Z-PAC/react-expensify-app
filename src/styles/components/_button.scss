.button {
  color: white;
  background: $blue;
  border: none;
  text-align: center;
  font-weight: 300;
  font-size: $font-size-large;
  justify-content: center;
  padding: $s-size;
  display: inline-block;
  text-decoration: none;
  line-height: 1;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888888;
}

.button--darkened {
  background: darken($color: $blue, $amount: 20%);
}