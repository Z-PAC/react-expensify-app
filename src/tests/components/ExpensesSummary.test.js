import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render a single expense summary correctly',() => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={500} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render multiple expenses summary correctly',() => {
  const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={1500} />);
  expect(wrapper).toMatchSnapshot();
});