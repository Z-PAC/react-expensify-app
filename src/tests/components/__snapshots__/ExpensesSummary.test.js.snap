// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a single expense summary correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        €5,00
      </span>
    </h1>
    <h2
      className="page-header__subtitle"
    >
       
      expenses
       hidden
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render multiple expenses summary correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totalling 
      <span>
        €15,00
      </span>
    </h1>
    <h2
      className="page-header__subtitle"
    >
       
      expenses
       hidden
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
